manual/Makefile

```diff
@@ -271,6 +271,8 @@ references.bib: $(addsuffix .tex, $(Targets))
 		$(LaTeX) $$i.tex >>.log; \
 		cat $$i.aux | grep -e "\(citation\|bibdata\|bibstyle\)" | sed 's/bibdata{references}/bibdata{$(BibFiles)}/g' >> all_refs.aux; \
 	done
+	# make glossary
```

Comment:
@Indanz manual.ist is generating a pdf build error: https://github.com/seL4/seL4/actions/runs/7079820384/job/19266878821?pr=1143

It runs locally but not in the PR checks.

Can you pls help fix this?

Reply:
Looking at the Makefile, I'd put this command somewhere after "LaTeX first pass" in the `%.pdf` target, either directly before or after the `BibTex` run.

Reply:
Thanks @lsf37! That works, at least locally. But it needs to runs of `make` for the glossary to be included in the pdf.

Reply:
The others check the log to see if a rerun is necessary, if you can check `makeindex`'s ouput to see if a rerun is necessary, that would be ideal. Otherwise add a manual rerun to the same section.

Reply:
For me it seems to work with:

```diff
diff --git a/manual/Makefile b/manual/Makefile
index bc17ce8e1..0bce8e7bf 100644
--- a/manual/Makefile
+++ b/manual/Makefile
@@ -228,6 +228,7 @@ clean:
                *.out *.ps *-diff.tex *.mps .log *.pdf *.tgz *~ *.lof *.lot env.tex
        rm -rf ${DoxygenOutput} ${GeneratedLatexDir} ${GeneratedMarkdownDir}
        rm -rf ${Stage}
+       rm -rf *.acn *.glo *.gls *.ilg *.ist
 
 tar:   clean
        ( p=`pwd` && d=`basename "$$p"` && cd .. && \
@@ -271,8 +272,6 @@ references.bib: $(addsuffix .tex, $(Targets))
                $(LaTeX) $$i.tex >>.log; \
                cat $$i.aux | grep -e "\(citation\|bibdata\|bibstyle\)" | sed 's/bibdata{references}/bibdata{$(BibFiles)}/g' >> all_refs.aux; \
        done
-       # make glossary
-       makeindex -s manual.ist -o manual.gls manual.glo
        @echo "====> Removing duplicate bib entries";
        ${Q}cat all_refs.aux | uniq > tmp.aux;
        ${Q}diff references.aux tmp.aux > references.diff 2> /dev/null; \
@@ -293,6 +292,7 @@ references.bib: $(addsuffix .tex, $(Targets))
        ${Q}if ! test -e $*.bbl || test $(Bib) -nt $*.bbl; then rm -f $*.bbl; fi
        @echo "====> LaTeX first pass: $(<)"
        ${Q}$(LaTeX) $< >.log || if egrep -q $(Error) $*.log ; then cat .log; rm $@; false ; fi
+       makeindex -s manual.ist -o manual.gls manual.glo # glossary
        ${Q}if egrep -q $(Rerun_Bib) $*.log ; then echo "====> BibTex" && $(BibTeX) $* > /dev/null && echo "====> LaTeX BibTeX pass" && $(LaTeX) >.log $< ; fi
        ${Q}if egrep -q $(Rerun) $*.log ; then echo "====> LaTeX rerun" && $(LaTeX) >.log $<; fi
        ${Q}if egrep -q $(Rerun) $*.log ; then echo "====> LaTeX rerun" && $(LaTeX) >.log $<; fi
```
That said, for me the generated pdf looks awful (unrelated to the addition of the glossary, probably wrong `roboto.sty`) and I don't see any references to the glossary definitions, but I don't know if there were supposed to be any.

Reply:
When I remove
`makeindex -s manual.ist -o manual.gls manual.glo`
after
`done`
(~line 274), the Glossary disappears from the TOC.

Also, there are no references to glossary items within the text. I can create them, but it will take some time.